docs/modules/ROOT/pages/adr/0043-appcat-framework-production-architecture.adoc

```diff
@@ -0,0 +1,1755 @@
+= ADR - AppCat Framework Production Architecture
+:adr_author: Gabriel Saratura, Claude (AI Assistant)
+:adr_owner: Schedar
+:adr_reviewers: Schedar
+:adr_date: 2024-12-02
+:adr_upd_date: 2024-12-02
+:adr_status: draft
+:adr_tags: architecture,crossplane,kcl,appcat,framework
+
+[NOTE]
+.Summary
+====
+This ADR defines the production architecture for the AppCat framework with the vision "Helm Chart in - Managed Service out".
+The architecture establishes clear separation between Framework Engineers (Go + KCL core) and Service Maintainers (service configs), leverages existing AppCat capabilities (SLI exporter, BillingService → Odoo, API Server), and provides comprehensive observability for Service Users via the Servala portal.
+
+**Key Decisions:**
+
+* *Configuration Language:* KCL over CUE for type-safe, constraint-based configuration
+* *Orchestration:* Crossplane 2.0 composition functions for extensible service management
+* *Secret Management:* Continue with Crossplane connection secrets
+* *Cost Tracking:* Keep existing BillingService CRD → Odoo integration
+* *Maintenance:* Composition revisions with CronJobs (MVP), investigate Argo Workflows and Crossplane Operations (Phase 2)
+* *Testing:* Chainsaw end-to-end tests + Go unit tests + optional golden tests
+
+====
+
+== Context
+
+=== Vision
+
+Transform any Kubernetes workload (Helm chart) into a standardized managed service with full lifecycle management - "Helm Chart in - Managed Service out".
+
+=== Requirements
+
+**Personas:**
+
+1. *Service User*: End customer using services via Servala portal (metrics, logs, costs, connection details)
+2. *Service Maintainer*: Adds new services, defines plans and Helm mappings (KCL only, no Go)
+3. *Service Operator*: Manages service instances, troubleshoots issues, performs maintenance
+4. *Framework Engineer*: Maintains platform infrastructure, composition functions, KCL framework
+
+**Key Constraints:**
+
+* Kubernetes API-based interactions
+* Open source tools
+* Automation-first approach
+* Single-tier clusters
+* Leverage existing: Servala portal, SLI exporter, API Server, BillingService, K8up backup, Crossplane
```

Comment:
This needs to be referenced to their source (docs and / or code) to be very clear